src/commands/cmd_config.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2018-2020 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Redis Labs Source Available License Agreement
+ */
+
+#include "cmd_config.h"
+#include "config.h"
+#include <string.h>
+
+void _Config_get_all(RedisModuleCtx *ctx) {
+	uint config_count = Config_END_MARKER;
+	RedisModule_ReplyWithArray(ctx, config_count);
+
+	for(Config_Option_Field field = 0; field < Config_END_MARKER; field++) {
+		long long value = 0;
+		const char *config_name = Config_Field_name(field);
+		bool res = Config_Option_get(field, &value);
+
+		if(!res || config_name == NULL) {
+			RedisModule_ReplyWithError(ctx, "Configuration field was not found");
+			return;
+		} else {
+			RedisModule_ReplyWithArray(ctx, 2);
+			RedisModule_ReplyWithCString(ctx, config_name);
+			RedisModule_ReplyWithLongLong(ctx, value);
+		}
+	}
+}
+
+void _Config_get(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+	// return the given config's value to the user
+	Config_Option_Field config_field;
+	const char *config_name = RedisModule_StringPtrLen(argv[2], NULL);
+
+	// return entire configuration
+	if(config_name[0] == '*') {
```

Comment:
I think an actual `strcmp` is better, as redis supports wildcard matches like `*max`.